src/companies/components/RegisterCompanyForm.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div>
+    <v-form ref="createCompanyForm" v-model="valid" lazy-validation>
+      <v-card>
+        <v-card-title> </v-card-title>
+
+        <v-text-field
+          v-model="companyInfo.name"
+          :rules="nameRules"
+          label="Name"
+          required
+          outlined
+        ></v-text-field>
+
+        <v-text-field
+          v-model="companyInfo.phoneNumber"
+          label="Phone number"
+          type="phone"
+          :rules="phoneNumberRules"
+          required
+          outlined
+        ></v-text-field>
+
+        <v-text-field
+          v-model="companyInfo.website"
+          label="Website"
+          :rules="websiteRules"
+          required
+          outlined
+        ></v-text-field>
+
+        <v-text-field
+          v-model="companyInfo.address"
+          label="Address"
+          :rules="addressRules"
+          required
+          outlined
+        ></v-text-field>
+
+        <v-select
+          v-model="companyInfo.currency"
+          :items="getCurrency"
+          label="Currency"
+          :rules="currencyRules"
+          dense
+          outlined
+        ></v-select>
+      </v-card>
+
+      <v-btn color="#72418b" dark @click="onSubmit()" x-large
+        >REGISTER A TEST COMPANY</v-btn
+      >
+    </v-form>
+  </div>
+</template>
+
+<script>
+import { currencyCodes } from "./currency.js";
+export default {
+  props: ["e1"],
+  data() {
+    return {
+      currencyCodes: currencyCodes,
+      currency: [],
+      companyInfo: {
+        name: "",
+        phoneNumber: "",
+        website: "",
+        address: "",
+        currency: "",
+      },
+      valid: true,
+      nameRules: [
+        (v) => !!v || "Name is required",
+        (v) => (v && v.length <= 20) || "Name must be less than 20 characters",
+      ],
+      websiteRules: [(v) => !!v || "Website is required"],
+      phoneNumberRules: [
+        (v) => !!v || "Phone number is required",
+        (v) =>
+          (v && v.length <= 10) ||
+          "Phone number must be less than 11 characters",
+      ],
+      addressRules: [(v) => !!v || "Address is required"],
+      currencyRules: [(v) => !!v || "Currency is required"],
+    };
+  },
+  methods: {
+    onSubmit() {
+      this.$emit("register-company", this.companyInfo);
+    },
+    validate() {
+      this.$refs.createCompanyForm.validate();
+    },
+    reset() {
+      this.$refs.form.reset();
+    },
+    resetValidation() {
+      this.$refs.form.resetValidation();
+    },
+  },
+  computed: {
+    getCurrency() {
+      this.currencyCodes.map((item) => {
+        this.currency.push(item.cc);
+      });
+      return this.currency;
+    },
```

Comment:
the computed property name shouldn't a `verb`
```suggestion
    currencies() {
      return this.currencyCodes.map((item) => item.cc);
    },
```

Reply:
And we don't need `currency` as a data property